cityofaustin/janis
# 4358 Janis Search: Pagination

ZenHub Issues:
- Search Pagination: https://app.zenhub.com/workspaces/techstack-5a78b88e1ce69f3510b678ef/issues/cityofaustin/techstack/4358
- Search Navigation:https://app.zenhub.com/workspaces/techstack-5a78b88e1ce69f3510b678ef/issues/cityofaustin/techstack/4283

Zenhub Issue: https://app.zenhub.com/workspaces/techstack-5a78b88e1ce69f3510b678ef/issues/cityofaustin/techstack/4358

Deployed: https://janis-v3-search-pag.netlify.com/

# Description

This PR will 
- Add Pagination to search
  - Also cleaned up the reusability of the component. There are some of @briaguya event page. changes in here as well. 
- Add hash query helper functions for cross-page uses.
  - I left notes in the code about use-cases. And, a bit of a breakdown of what's going on.
  - Let me know if i can add or clarify in anyway.
- Add Search icon to navigation


# Types of changes

  - Back from pagination was blocking history. That's fixed now

# How can this be tested?

👇 https://janis-v3-search-pag.netlify.com/
- hit the search button 
- search something, hit the next page on pagination
- go back in pagination.
- search "official"
- Click the only result. 
- paginate some more
- go back. 

NOTE: You may notice having to press back twice after returning back through the pagination. This is because `/search` registers to history then `/search#page=1`. I had some ideas to fix, but felt it was ok for MVP and maybe not even that "wrong" in what you'd expect. 

# Checklist:
